src/mlpack/tests/main_tests/softmax_regression_test.cpp

```diff
@@ -0,0 +1,513 @@
+
+  // Train SR for lamda 0.9.
+
+  // Input training data.
+  SetInputParam("training", std::move(inputData2));
+  SetInputParam("labels", std::move(labels2));
+  SetInputParam("max_iterations", (int) 1000);
+  SetInputParam("test", std::move(testData2));
+
+  mlpackMain();
+
+  // Check that initial parameters and final parameters matrix
+  // using saved model are different.
+  bool flag = false;
+  bool* flagPtr = &flag;
+  for (size_t i = 0; i < modelParam.n_elem; ++i)
+  {
+    if ((int) (modelParam[i] * 1e+6) != (int) (CLI::GetParam<SoftmaxRegression>
+                                      ("output_model").Parameters()[i] * 1e+6))
+    {
+      *flagPtr = true;
+      break;
+    }
+  }
+  BOOST_REQUIRE_EQUAL(flag, true);
+}
+
+/**
+ * Check that output object parameter for no_intercept
+ * term is one less than with intercept term.
+ */
+BOOST_AUTO_TEST_CASE(SoftmaxRegressionDiffInterceptTest)
+{
+  // Train SR with intercept.
+  arma::mat inputData;
+  if (!data::Load("trainSet.csv", inputData))
+    BOOST_FAIL("Cannot load train dataset trainSet.csv!");
+
+  // Get the labels out.
+  arma::Row<size_t> labels(inputData.n_cols);
+  for (size_t i = 0; i < inputData.n_cols; ++i)
+    labels[i] = inputData(inputData.n_rows - 1, i);
+
+  // Delete the last row containing labels from input dataset.
+  inputData.shed_row(inputData.n_rows - 1);
+
+  arma::mat testData;
+  if (!data::Load("testSet.csv", testData))
+    BOOST_FAIL("Cannot load test dataset testSet.csv!");
+
+  // Delete the last row containing labels from test dataset.
+  testData.shed_row(testData.n_rows - 1);
+
+  size_t testSize = testData.n_cols;
+
+  // Create a copy of data to be reused.
+  arma::mat inputData2 = inputData;
+  arma::mat testData2 = testData;
+  arma::Row<size_t> labels2 = labels;
+
+  // Input training data.
+  SetInputParam("training", std::move(inputData));
+  SetInputParam("labels", std::move(labels));
+  SetInputParam("no_intercept", (bool) true);
+
+  // Input test data.
+  SetInputParam("test", std::move(testData));
+
+  mlpackMain();
+
+  // Store output parameters.
+  arma::mat modelParam;
+  modelParam = CLI::GetParam<SoftmaxRegression>("output_model").Parameters();
+
+  // Reset passed parameters.
+  CLI::GetSingleton().Parameters()["training"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["labels"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["no_intercept"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["test"].wasPassed = false;
+
+  // Train SR for no_intercept.
+
+  // Input training data.
+  SetInputParam("training", std::move(inputData2));
+  SetInputParam("labels", std::move(labels2));
+  SetInputParam("test", std::move(testData2));
+
+  mlpackMain();
+
+  // Check that initial parameters has 1 more parameter than
+  // final parameters matrix.
+  BOOST_REQUIRE_EQUAL(
+    CLI::GetParam<SoftmaxRegression>("output_model").Parameters().n_cols,
+    modelParam.n_cols + 1);
```

Comment:
A minor style issue---the wrapped lines here should be indented two tabs from the first line, not one. :)